File: app/src/main/java/de/nulide/shiftcal/data/legacy/model/MonthNote.java
package de.nulide.shiftcal.data.legacy.model;

public class MonthNote {
    private int year;
    private int month;
    private String note;

    public MonthNote() {
        this.year = 1990;
        this.month = 1;
        this.note = "";
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public int getMonth() {
        return month;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }
}

File: app/src/main/java/de/nulide/shiftcal/data/model/MonthNote.kt
package de.nulide.shiftcal.data.model

import androidx.room.Entity
import androidx.room.ForeignKey

@Entity(
    tableName = "month_note",
    primaryKeys = ["calendarId", "year", "month"],
    foreignKeys = [
        ForeignKey(
            entity = ShiftCalendar::class,
            parentColumns = ["id"],
            childColumns = ["calendarId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class MonthNote(
    var calendarId: Int,
    val year: Int,
    val month: Int,
    var msg: String
) {
    constructor(year: Int, month: Int, msg: String) : this(0, year, month, msg)

    constructor() : this(0, 0, 0, "")
}